tools/test.py

```diff
@@ -4,21 +4,23 @@
 import os
 import sys
 from check_output_test import check_output_test
-from util import executable_suffix, run
+from util import executable_suffix, run, build_path
 
 
 def check_exists(filename):
     if not os.path.exists(filename):
         print "Required target doesn't exist:", filename
-        print "Build target :all"
+        print "Run ./tools/build.py"
         sys.exit(1)
 
 
 def main(argv):
-    if len(argv) != 2:
-        print "Usage: tools/test.py [build dir]"
+    build_dir = build_path()
```

Comment:
I'd find it easier to reason about written like this:

```python
if len(argv) == 2:
  build_dir = sys.argv[1]
elif len(argv) == 1:
  build_dir = build_path()
else:
  print "Usage: tools/test.py [build_dir]"
  sys.exit(1)
```

Reply:
Changed.